common/bomb.cpp

```diff
@@ -59,6 +58,45 @@ bomb::update()
   return;
 }
 
+std::vector<std::vector<TargetSquare>>
+bomb::identifyTargetSquares()
+{
+  /** Create stacks of squares that can be iterated in a FIFO fashion **/
+  std::vector<std::vector<TargetSquare>> targets;
+  int i = int(mPosition[0]); int j = int(mPosition[1]);
+
+  auto makeSquare = [](int a, int b){
+    return std::make_shared<actor>(a, b, 1, 1, false);
+  };
+
+  // Square bomb is sat on
+  targets.push_back({makeSquare(i, j)});
+
+  // Horizontal
+  std::vector<TargetSquare> left;
+  std::vector<TargetSquare> right;
+  for(int power = 1; power <= mPower; power++) {
+    left.push_back(makeSquare(i - power, j));
+    right.push_back(makeSquare(i + power, j));
+  }
+
+  // Vertical
+  std::vector<TargetSquare> up;
+  std::vector<TargetSquare> down;
+  for(int power = 1; power <= mPower; power++) {
+    up.push_back(makeSquare(i, j + power));
+    down.push_back(makeSquare(i, j - power));
+  }
+
+  targets.push_back(left);
+  targets.push_back(right);
+  targets.push_back(up);
+  targets.push_back(down);
```

Comment:
```suggestion
  for(int power = 1; power <= mPower; power++) {
    targets.push_back(makeSquare(i - power, j)); //Left
    targets.push_back(makeSquare(i + power, j)); //Right
    targets.push_back(makeSquare(i, j + power)); //Up
    targets.push_back(makeSquare(i, j - power)); //Down
 }
```

Reply:
Please check if sane, but it seems we don't need these unnecessary copies

Reply:
Note targets is `vec<vec<Square>>` not `vec<Square>`. This makes traversal simpler in the destruction logic below, (the destruction 'paths' in each direction behave independently).

Reply:
Perhaps we could introduce a `DestructionPath` struct to make this more explicit. Can do if you think it makes it more readable

Reply:
No need 👍